include/unifex/find_if.hpp

```diff
@@ -0,0 +1,205 @@
+
+private:
+
+  template <typename BeginIt, typename EndIt, typename... Args>
+  using result = type_list<type_list<BeginIt, Args...>>;
+
+public:
+
+  template <
+      template <typename...> class Variant,
+      template <typename...> class Tuple>
+  using value_types = type_list_nested_apply_t<
+    typename Predecessor::template value_types<concat_type_lists_unique_t, result>,
+    Variant,
+    Tuple>;
+
+  template <template <typename...> class Variant>
+  using error_types = typename concat_type_lists_unique_t<
+    typename Predecessor::template error_types<type_list>,
+    type_list<std::exception_ptr>>::template apply<Variant>;
+
+  template <typename Receiver>
+  using receiver_t = receiver_t<Receiver, Func, FuncPolicy>;
+
+  friend constexpr auto tag_invoke(tag_t<blocking>, const type& sender) {
+    return blocking(sender.pred_);
+  }
+
+  template(typename Sender, typename Receiver)
+    (requires same_as<remove_cvref_t<Sender>, type> AND receiver<Receiver>)
+  friend auto tag_invoke(tag_t<unifex::connect>, Sender&& s, Receiver&& r)
+    noexcept(
+      std::is_nothrow_constructible_v<remove_cvref_t<Receiver>, Receiver> &&
+      std::is_nothrow_constructible_v<Func, decltype((static_cast<Sender&&>(s).func_))> &&
+      is_nothrow_connectable_v<decltype((static_cast<Sender&&>(s).pred_)), receiver_t<remove_cvref_t<Receiver>>>)
+      -> connect_result_t<decltype((static_cast<Sender&&>(s).pred_)), receiver_t<remove_cvref_t<Receiver>>> {
+    return unifex::connect(
+      static_cast<Sender&&>(s).pred_,
+      receiver_t<remove_cvref_t<Receiver>>{
+        static_cast<Sender&&>(s).func_,
+        static_cast<Receiver&&>(r),
+        static_cast<Sender&&>(s).funcPolicy_});
+  }
+};
+} // namespace _find_if
+
+namespace _find_if_cpo {
+  inline const struct _fn {
+  private:
+    template <typename Sender, typename Func, typename FuncPolicy>
+    using _result_t =
+      typename conditional_t<
+        tag_invocable<_fn, Sender, Func>,
+        meta_tag_invoke_result<_fn>,
+        meta_quote3<_find_if::sender>>::template apply<Sender, Func, FuncPolicy>;
+  public:
+    template(typename Sender, typename Func, typename FuncPolicy)
+      (requires tag_invocable<_fn, Sender, Func, FuncPolicy>)
+    auto operator()(Sender&& predecessor, Func&& func, FuncPolicy policy) const
+        noexcept(is_nothrow_tag_invocable_v<_fn, Sender, Func, FuncPolicy>)
+        -> _result_t<Sender, Func, FuncPolicy> {
+      return unifex::tag_invoke(_fn{}, (Sender&&)predecessor, (Func&&)func, (FuncPolicy&&)policy);
+    }
+    template(typename Sender, typename Func, typename FuncPolicy)
+      (requires (!tag_invocable<_fn, Sender, Func, FuncPolicy>))
+    auto operator()(Sender&& predecessor, Func&& func, FuncPolicy policy) const
+        noexcept(std::is_nothrow_constructible_v<
+          _find_if::sender<Sender, Func, FuncPolicy>, Sender, Func, FuncPolicy>)
+        -> _result_t<Sender, Func, FuncPolicy> {
```

Comment:
And similarly, can the return type here just be `_find_if::sender<Sender, Func, FuncPolicy>`?